python/ray/data/tests/test_formats.py

```diff
@@ -162,6 +166,33 @@ def test_write_datasource(ray_start_regular_shared):
     assert ray.get(output.data_sink.get_rows_written.remote()) == 10
 
 
+def test_open_file_with_retry(ray_start_regular_shared):
+    class Counter:
```

Comment:
rename it to something like FlakyFileOpener? also `max_attempts` can be set in `__init__`.

Reply:
updated.